src/AzureExtension/Services/DevBox/DevBoxManagementService.cs

```diff
@@ -120,6 +129,31 @@ public async Task<string> HttpsRequestToDataPlaneWithRawResponse(HttpClient clie
         }
     }
 
+    // Filter out projects that the user does not have write access to
+    private bool DeveloperHasWriteAbility(DevBoxProject project, IDeveloperId developerId)
+    {
+        try
+        {
+            var uri = $"{project.Properties.DevCenterUri}{Constants.Projects}/{project.Name}/users/me/abilities?{Constants.APIVersion}";
+            var result = HttpsRequestToDataPlane(new Uri(uri), developerId, HttpMethod.Get, null).Result;
+            var rawResponse = result.JsonResponseRoot.ToString();
+            var abilities = JsonSerializer.Deserialize<AbilitiesJSONToCSClasses.BaseClass>(rawResponse, _jsonOptions);
+            _log.Debug($"Response from abilities: {rawResponse}");
+
+            if (abilities!.AbilitiesAsDeveloper.Contains(_writeAbility) || abilities!.AbilitiesAsAdmin.Contains(_writeAbility))
+            {
+                return true;
+            }
+
+            return false;
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, $"Unable to get abilities for {project.Name}");
+            return true;

```

Comment:
Heh, I wasn't thinking about this weirdness being true now - but makes sense.  :shrug: